### PR TITLE
perl: Use local include path

### LINKED ISSUE #405: perl 5.22.1: Cannot find errno.h at Errno_pm.PL line 131.
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports:

When trying to build perl on a Chrome OS with Linuxbrew I get "Cannot find errno.h at Errno_pm.PL line 131." and the build fails. Something must have changed with the formula (or Linuxbrew) because I *was* able to build perl before in this very same machine.

Here's the log file:

https://gist.github.com/anonymous/62a0bf3e1706bb0540727bac0bd80e62

### PR DESCRIPTION
**Pull request by DoomHammer**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

-----

Should fix #405

### CODE REVIEW

File: Formula/perl.rb [hunk 1]
```diff
@@ -35,6 +35,7 @@ def install
 
     args << "-Dusedtrace" if build.with? "dtrace"
     args << "-Dusedevel" if build.head?
+    args << "-Dlocincpth=#{HOMEBREW_PREFIX}/includet dif" if OS.linux?
```
Comment (sjackman) [inline Formula/perl.rb]: Typo? `includet dif`

File: Formula/perl.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ class Perl < Formula
   url "http://www.cpan.org/src/5.0/perl-5.24.0.tar.xz"
   mirror "https://mirrors.ocf.berkeley.edu/debian/pool/main/p/perl/perl_5.24.0.orig.tar.xz"
   sha256 "a9a37c0860380ecd7b23aa06d61c20fc5bc6d95198029f3684c44a9d7e2952f2"
-  revision 1
+  revision 2
```
Comment (sjackman) [inline Formula/perl.rb]: This'll have to become
```ruby
revision OS.linux? ? 2 : 1
```

Comment (DoomHammer) [inline Formula/perl.rb]: Don't know about `brew audit`, but Ruby doesn't like it.

**Comment by DoomHammer** [comment]
No problems running `brew test-bot` locally. Should I merge this?

**Comment by sjackman** [comment]
Could you please post a small test case that reproduces the issue that this PR fixes?
Could you add a comment to the formula that explains what issue this PR fixes?

**Comment by DoomHammer** [comment]
I can add a comment to the formula, but test-case would probably be far from small. You would need a system without libc headers (like a Chromebook or busybox-based one) and those still come with their own set of problems (eg. plain installation of Linuxbrew usually fails).

**Comment by sjackman** [comment]
Does this issue manifest in some other way, for example using the wrong headers when compiling `perl` modules with `cpanm`? A test case would be nice, but if it's not easy, go ahead and merge.

**Comment by DoomHammer** [comment]
As far as I know it is only used during compilation of `perl` itself. Specifically: to generate a Perl module with proper errno consts.

**Comment by sjackman** [comment]
Cool. Good to know.

**Comment by DoomHammer** [comment]
Now this is funny (as in: not)! Travis says `brew test --verbose vim` fails. Neither CircleCI nor local env can reproduce the same.

On the other hand CircleCI fails with now famous `Error: --keep-old was passed but there were changes in sha256 => x86_64_linux!`.

**Comment by sjackman** [comment]
```
/home/linuxbrew/.linuxbrew/Cellar/vim/7.4.2085/bin/vim
--version
/home/linuxbrew/.linuxbrew/Cellar/vim/7.4.2085/bin/vim: error while loading shared libraries: libperl.so: cannot open shared object file: No such file or directory
Error: vim: failed
```

Does perl install `libperl.so` in `lib`?

**Comment by rwhogg** [comment]
> Does perl install libperl.so in lib? I wonder why vim can't find it.

No, it doesn't. At least on my local, `libperl.so` can be found in `$HOMEBREW_PREFIX/opt/perl/lib/perl5/5.24.0/x86_64-linux-thread-multi/CORE/libperl.so`.

There have been other instances of being unable to link against libperl, such as #128

**Comment by sjackman** [comment]
How is this supposed to work on other systems, e.g. Debian and Ubuntu? Do they add the above directory to `/etc/ld.so.conf`?

**Comment by rwhogg** [comment]
> How is this supposed to work on other systems, e.g. Debian and Ubuntu? Do they add the above directory to /etc/ld.so.conf?

Linuxbrew's Perl doesn't link into `$HOMEBREW_PREFIX/lib`, but on Ubuntu 14.04, `/usr/lib/libperl.so` (and a few variants) does exist. There's no mention of Perl in /etc/ld.so.conf.d/

**Comment by sjackman** [comment]
On my system with `vim` built from source, `/gsc/btl/linuxbrew/opt/readline/lib:/gsc/btl/linuxbrew/Cellar/perl/5.24.0_1/lib/perl5/5.24.0/x86_64-linux-thread-multi/CORE` is in the RPATH of `vim`. Rebuilding the `vim` bottle *may* fix this issue.

**Comment by sjackman** [comment]
@DoomHammer If you don't expect this patch to affect the functionality of the bottle, I would suggest reverting the revision change and merging this PR without updating the bottle.

**Comment by DoomHammer** [comment]
This should not affect the bottle. It would only help when building from source on systems lacking glibc headers. I will do as suggested.

**Comment by DoomHammer** [comment]
Have any of you, @sjackman, @rwhogg have been playing tricks with either Travis or `vim`? Now the build magically succeeded...

**Comment by rwhogg** [comment]
> Have any of you, @sjackman, @rwhogg have been playing tricks with either Travis or vim? Now the build magically succeeded...

I haven't done anything with Vim or with the homebrew-core Travis, so not sure what's up.

**Comment by sjackman** [comment]
Me neither.

**Comment by sjackman** [comment]
The `master` build will succeed where the pull request failed, if that's what you're seeing.

**Pull request closed without merging**